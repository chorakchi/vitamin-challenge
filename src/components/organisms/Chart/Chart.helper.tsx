  export const optionsProvider = (items : any) => ({
    chart: {
      plotBackgroundColor: null,
      plotBorderWidth: null,
      plotShadow: false,
      type: 'spline',
      zoom:'xy'
    },
    title:'',
    xAxis: {
    	type: "datetime",
      title: {
        text: ''        
      },
      labels: {
        formatter: function() {
          return  this.y;
        }
      }
    }, 
    yAxis: {
      title: {
        text: ''        
      },
      labels: {
        formatter: function() {
          return  this.x;
        }
      }
    }, 
  


  series: items
  })