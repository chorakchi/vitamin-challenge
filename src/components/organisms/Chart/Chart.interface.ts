import { HTMLAttributes } from "react";

export interface ChartProps extends HTMLAttributes<HTMLElement> {
  items: any;
}
