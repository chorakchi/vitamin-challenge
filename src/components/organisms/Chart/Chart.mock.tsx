export const mockData = {
  items: [
    {
        "name": "EAT",
        "data": [
            [
                "2017-01-03",
                67.5
            ],
            [
                "2017-02-01",
                127.64
            ],
            [
                "2017-03-01",
                190.57
            ],
            [
                "2017-04-03",
                265.06
            ],
            [
                "2017-05-01",
                334.26
            ],
            [
                "2017-06-01",
                373.46
            ],
            [
                "2017-07-03",
                423.84
            ],
            [
                "2017-08-01",
                459.62
            ],
            [
                "2017-09-01",
                470.72
            ],
            [
                "2017-10-02",
                542.14
            ],
            [
                "2017-11-01",
                633.73
            ],
            [
                "2017-12-01",
                761.52
            ],
            [
                "2018-01-02",
                901.52
            ],
            [
                "2018-02-01",
                911.45
            ],
            [
                "2018-03-01",
                919.48
            ],
            [
                "2018-04-02",
                1023.46
            ],
            [
                "2018-05-01",
                1346.69
            ],
            [
                "2018-06-01",
                1375.17
            ],
            [
                "2018-07-02",
                1584.39
            ],
            [
                "2018-08-01",
                1546.72
            ],
            [
                "2018-09-04",
                1571.82
            ],
            [
                "2018-10-01",
                1775.11
            ],
            [
                "2018-11-01",
                1708.57
            ],
            [
                "2018-12-03",
                2025.55
            ],
            [
                "2019-01-02",
                1878.12
            ],
            [
                "2019-02-01",
                1797.54
            ],
            [
                "2019-03-01",
                2043.67
            ],
            [
                "2019-04-01",
                2087.34
            ],
            [
                "2019-05-01",
                2018.29
            ],
            [
                "2019-06-03",
                1904.14
            ],
            [
                "2019-07-01",
                2009.51
            ],
            [
                "2019-08-01",
                2140.42
            ],
            [
                "2019-09-03",
                2097.73
            ],
            [
                "2019-10-01",
                2391.28
            ],
            [
                "2019-11-01",
                2629.14
            ],
            [
                "2019-12-02",
                2730.92
            ],
            [
                "2020-01-02",
                2585.78
            ],
            [
                "2020-02-03",
                2572.55
            ],
            [
                "2020-03-02",
                2242.49
            ],
            [
                "2020-04-01",
                811.54
            ],
            [
                "2020-05-01",
                1612.89
            ],
            [
                "2020-06-01",
                2074.77
            ],
            [
                "2020-07-01",
                1829.11
            ],
            [
                "2020-08-03",
                2122.87
            ],
            [
                "2020-09-01",
                3814.81
            ],
            [
                "2020-10-01",
                3645.23
            ],
            [
                "2020-11-02",
                3826.3
            ],
            [
                "2020-12-01",
                4491.69
            ],
            [
                "2021-01-04",
                4760.65
            ],
            [
                "2021-02-01",
                5488.31
            ],
            [
                "2021-03-01",
                6581.33
            ],
            [
                "2021-04-01",
                6603.38
            ],
            [
                "2021-05-03",
                6052.88
            ],
            [
                "2021-06-01",
                5393.25
            ]
        ]
    },
    {
        "name": "CAKE",
        "data": [
            [
                "2017-01-03",
                292.5
            ],
            [
                "2017-02-01",
                582.79
            ],
            [
                "2017-03-01",
                897.12
            ],
            [
                "2017-04-03",
                1218.8
            ],
            [
                "2017-05-01",
                1523.26
            ],
            [
                "2017-06-01",
                1705.93
            ],
            [
                "2017-07-03",
                1733.61
            ],
            [
                "2017-08-01",
                1941.53
            ],
            [
                "2017-09-01",
                1978.53
            ],
            [
                "2017-10-02",
                2319.4
            ],
            [
                "2017-11-01",
                2758.78
            ],
            [
                "2017-12-01",
                3237.04
            ],
            [
                "2018-01-02",
                3615.81
            ],
            [
                "2018-02-01",
                4024.17
            ],
            [
                "2018-03-01",
                4016.97
            ],
            [
                "2018-04-02",
                4341.03
            ],
            [
                "2018-05-01",
                5117.28
            ],
            [
                "2018-06-01",
                5339.95
            ],
            [
                "2018-07-02",
                6014.17
            ],
            [
                "2018-08-01",
                5584.02
            ],
            [
                "2018-09-04",
                6312.45
            ],
            [
                "2018-10-01",
                6557.21
            ],
            [
                "2018-11-01",
                6362.51
            ],
            [
                "2018-12-03",
                6449.43
            ],
            [
                "2019-01-02",
                6206.99
            ],
            [
                "2019-02-01",
                6765.79
            ],
            [
                "2019-03-01",
                7341.14
            ],
            [
                "2019-04-01",
                7906.29
            ],
            [
                "2019-05-01",
                8272.01
            ],
            [
                "2019-06-03",
                7695.82
            ],
            [
                "2019-07-01",
                7849.8
            ],
            [
                "2019-08-01",
                8349.92
            ],
            [
                "2019-09-03",
                7596.8
            ],
            [
                "2019-10-01",
                8423.23
            ],
            [
                "2019-11-01",
                8825.55
            ],
            [
                "2019-12-02",
                9534.5
            ],
            [
                "2020-01-02",
                8748.57
            ],
            [
                "2020-02-03",
                8807.59
            ],
            [
                "2020-03-02",
                8661.22
            ],
            [
                "2020-04-01",
                4070.37
            ],
            [
                "2020-05-01",
                5504.75
            ],
            [
                "2020-06-01",
                5975
            ],
            [
                "2020-07-01",
                6546.41
            ],
            [
                "2020-08-03",
                7290.38
            ],
            [
                "2020-09-01",
                9934.62
            ],
            [
                "2020-10-01",
                9532.5
            ],
            [
                "2020-11-02",
                10261.63
            ],
            [
                "2020-12-01",
                13590.42
            ],
            [
                "2021-01-04",
                12708.31
            ],
            [
                "2021-02-01",
                16428.22
            ],
            [
                "2021-03-01",
                20557.38
            ],
            [
                "2021-04-01",
                22191.71
            ],
            [
                "2021-05-03",
                23539.7
            ],
            [
                "2021-06-01",
                22059.78
            ]
        ]
    },
    {
        "name": "PZZA",
        "data": [
            [
                "2017-01-03",
                90
            ],
            [
                "2017-02-01",
                177.56
            ],
            [
                "2017-03-01",
                260.11
            ],
            [
                "2017-04-03",
                352.87
            ],
            [
                "2017-05-01",
                436.22
            ],
            [
                "2017-06-01",
                544.58
            ],
            [
                "2017-07-03",
                568.4
            ],
            [
                "2017-08-01",
                651.53
            ],
            [
                "2017-09-01",
                771.67
            ],
            [
                "2017-10-02",
                842.38
            ],
            [
                "2017-11-01",
                812.81
            ],
            [
                "2017-12-01",
                872.78
            ],
            [
                "2018-01-02",
                920.42
            ],
            [
                "2018-02-01",
                1117.87
            ],
            [
                "2018-03-01",
                1143.48
            ],
            [
                "2018-04-02",
                1212.91
            ],
            [
                "2018-05-01",
                1388.69
            ],
            [
                "2018-06-01",
                1217.58
            ],
            [
                "2018-07-02",
                1279.08
            ],
            [
                "2018-08-01",
                1170.04
            ],
            [
                "2018-09-04",
                1357.47
            ],
            [
                "2018-10-01",
                1550.64
            ],
            [
                "2018-11-01",
                1813.42
            ],
            [
                "2018-12-03",
                1638.83
            ],
            [
                "2019-01-02",
                1496.16
            ],
            [
                "2019-02-01",
                1525.76
            ],
            [
                "2019-03-01",
                1806.33
            ],
            [
                "2019-04-01",
                2254.1
            ],
            [
                "2019-05-01",
                2325.86
            ],
            [
                "2019-06-03",
                2288.9
            ],
            [
                "2019-07-01",
                2205.36
            ],
            [
                "2019-08-01",
                2246.33
            ],
            [
                "2019-09-03",
                2469.64
            ],
            [
                "2019-10-01",
                2877.16
            ],
            [
                "2019-11-01",
                3262.65
            ],
            [
                "2019-12-02",
                3625
            ],
            [
                "2020-01-02",
                3713.85
            ],
            [
                "2020-02-03",
                3887.82
            ],
            [
                "2020-03-02",
                3564.34
            ],
            [
                "2020-04-01",
                3551.24
            ],
            [
                "2020-05-01",
                4650.6
            ],
            [
                "2020-06-01",
                5264.99
            ],
            [
                "2020-07-01",
                5557.95
            ],
            [
                "2020-08-03",
                6534.96
            ],
            [
                "2020-09-01",
                7013.34
            ],
            [
                "2020-10-01",
                5841.62
            ],
            [
                "2020-11-02",
                5383.42
            ],
            [
                "2020-12-01",
                5884.78
            ],
            [
                "2021-01-04",
                6268.73
            ],
            [
                "2021-02-01",
                7556.07
            ],
            [
                "2021-03-01",
                7062.48
            ],
            [
                "2021-04-01",
                6752.64
            ],
            [
                "2021-05-03",
                7568.42
            ],
            [
                "2021-06-01",
                7457.61
            ]
        ]
    }
],
};
