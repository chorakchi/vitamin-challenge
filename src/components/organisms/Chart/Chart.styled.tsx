import styled from "styled-components";
import { Box } from "components";

export const StyledWrapper = styled(Box)`
@keyframes fade-in-animation {
    0% {
        opacity: 0;
    }
    100% {
        opacity: 1;
    }
}

animation: fade-in-animation 1s;
  position: relative;
  margin: 0;
  width: 100%;
  height: 100%;
  gap: 18px;
  @media screen and (max-width: 1280px) {
    gap: 0;
    padding: 2px;
    margin: 0;
  }
  display: ${({ hidden }) => (hidden ? "hidden" : "unset")};

  .highcharts-credits {
    display: none;
  }

  h3 {
    font-size: 28px;
    padding: 4px 0;
  }

  th {
    font-weight: bold;
    padding-right: 8px;
    color: gray;
  }
`;
