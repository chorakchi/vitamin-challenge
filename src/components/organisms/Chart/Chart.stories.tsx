import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Chart } from "./Chart";
import {mockData } from './Chart.mock'

export default {
  title: "Organisms/Chart",
  component: Chart,
} as ComponentMeta<typeof Chart>;

const Template: ComponentStory<typeof Chart> = (args) => (
  <Chart {...args} />
);

export const Chart_ = Template.bind({});
Chart_.args = {
  ...mockData
}