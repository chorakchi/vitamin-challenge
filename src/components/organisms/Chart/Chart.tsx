import { StyledWrapper } from "./Chart.styled";
import { ChartProps } from "./Chart.interface";
import Highcharts from "highcharts";
import HighchartsReact from "highcharts-react-official";
import HighchartMore from "highcharts/highcharts-more";
import React, { useState } from "react";
import { optionsProvider } from "./Chart.helper";

export const Chart: React.FC<ChartProps> = ({
  items,
  ...props
}: ChartProps) => {
  const [highchartMoreLoaded, setHighchartMoreLoaded] = useState(false);
  setTimeout(() => {
    HighchartMore(Highcharts);
    setHighchartMoreLoaded(true);
  }, 0);
console.log('dd',optionsProvider(items))
  return (
    <StyledWrapper hidden= {!items.length}>
      {highchartMoreLoaded && (
        <HighchartsReact {...props} highcharts={Highcharts} options={optionsProvider(items)} />
      )}
    </StyledWrapper>
  );
};
