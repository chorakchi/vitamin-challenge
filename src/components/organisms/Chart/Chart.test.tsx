import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Chart } from "./Chart";
import { mockData } from "./Chart.mock";

describe("Chart", () => {
  it("should render the Chart", () => {
    const subject = shallow(
      <Chart {...mockData}/>
    );
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer
      .create(
        <Chart {...mockData}/>
      )
      .toJSON();
    expect(subject).toMatchSnapshot();
  });
});
