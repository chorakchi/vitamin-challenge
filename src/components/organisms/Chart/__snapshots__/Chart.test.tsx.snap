// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart should render correctly and match to the Snapshot 1`] = `
<div
  className="Boxstyled__BoxStyled-sc-12nhtob-0 bOPHwF Chartstyled__StyledWrapper-sc-2zef0p-0 clZRma"
  hidden={false}
/>
`;
